server/fleet/teams.go

```diff
@@ -185,6 +188,11 @@ func (t *TeamMDM) Copy() *TeamMDM {
 	if t.MacOSSettings.DeprecatedEnableDiskEncryption != nil {
 		clone.MacOSSettings.DeprecatedEnableDiskEncryption = ptr.Bool(*t.MacOSSettings.DeprecatedEnableDiskEncryption)
 	}
+	if t.WindowsSettings.CustomSettings.Set {
+		windowsSettings := make([]string, len(t.WindowsSettings.CustomSettings.Value))
+		copy(windowsSettings, t.WindowsSettings.CustomSettings.Value)
+		t.WindowsSettings.CustomSettings = optjson.SetSlice[string](windowsSettings)
```

Comment:
Oops, that's the so tricky error of setting the copy to the original and not the clone!
```suggestion
		clone.WindowsSettings.CustomSettings = optjson.SetSlice[string](windowsSettings)
```